documentation/src/main/asciidoc/userguide/chapters/caching/Caching.adoc

```diff
@@ -559,6 +559,18 @@ In order to control which provider to use and specify configuration for the `Cac
 
 Only by specifying the second property `hibernate.javax.cache.uri` will you be able to have a `CacheManager` per `SessionFactory`.
 
+===== Using a non-default JCache `CacheManager`
+
+If you don't want to use the default `CacheManager`, you need to set the `hibernate.javax.cache.cache_manager` configuration property
+to one of the following values:
+
+Object reference:: If the value is an Object instance implementing the `CacheManager` interface,
+the provided `CacheManager` instance will be used.
+`Class`:: If the value is a Java `Class` object that implements the `CacheManager` interface,
+Hibernate will create a new instance for that `Class` and use it instead of the default one.
+`String`:: If the value is a Java `String`, Hibernate expects it to be the fully-qualified `Class` name
+of the `CacheManager` implementation which will be used to instantiate the non-default `CacheManager`.
```

Comment:
Looks good to me but I think for `Class` and `String`, we should be explicit about what we expect as a constructor in the referenced class.

Could you check and add that?

Thanks!

Reply:
@gsmet Are you asking what constructor is used when building the `CacheManager` instance? It's the default one, of course.

Reply:
I'll make it more obvious then, maybe that's going to be relevant when providing the `CacheManager` as a Spring bean, as mentioned in the Jira issue.